src/server/handlers/github.js

```diff
@@ -321,6 +322,20 @@ const getRequest = (owner, name, token, secret) => {
   };
 };
 
+
+export const hasRepoChanged = async (repositoryUrl, last_updated_at, token) => {
+  if (last_updated_at === undefined || !last_updated_at) {
+    throw new Error('`last_updated_at` must be given.');
+  }
+  const since = moment(last_updated_at).toISOString();
+  const { owner, name } = parseGitHubRepoUrl(repositoryUrl);
+  const uri = `/repos/${owner}/${name}/commits?since=${since}`;
+  const response = await requestGitHub.get(uri, { token, json: true });
```

Comment:
Good idea, added basic support for `If-Modified-Since` based on https://developer.github.com/v3/#conditional-requests.
We can continue to use a single global `last_updated_at` timestamp for snap repositories.